google/cloud/storage/retry.py

```diff
@@ -123,17 +121,11 @@ def is_metageneration_specified(query_params):
 def is_etag_in_json(data):
     """Return True if an etag is contained in the JSON body.
 
-    Indended for use on calls with relatively short JSON payloads."""
-    try:
-        content = json.loads(data)
-        if content.get("etag"):
-            return True
-    # Though this method should only be called when a JSON body is expected,
-    # the retry policy should be robust to unexpected payloads.
-    # In Python 3 a JSONDecodeError is possible, but it is a subclass of ValueError.
-    except (ValueError, TypeError):
-        pass
-    return False
+    :type data: dict or None
+    :param data: A dict containing the JSON body. If not passed, returns False.
```

Comment:
Nit: I think we could be clearer that the dict _represents_ the JSON body rather than containing it. Also, this is specifically intended for calls where metadata is sent in the request body.

It seems like it'd be better to actually just rename this function to `is_etag_in_data` but I guess technically this is part of the public surface of the library (even if we don't expect external callers to use it directly)?

Reply:
We could rename it and leave `is_etag_in_json` around as a backward-compatibility alias.

Reply:
Would it be something like this:

```
def is_etag_in_data(data):
    """Return True if an etag is contained in the request body.

    :type data: dict or None
    :param data: A dict representing the request JSON body. If not passed, returns False.
    """
    return data is not None and "etag" in data


def is_etag_in_json(data):
    """
    ``is_etag_in_json`` is supported for backwards-compatibility reasons only;
    please use ``is_etag_in_data`` instead.
    """
    return is_etag_in_data(data)
```

Reply:
LGTM